lib/omnibus-ctl.rb

```diff
@@ -506,6 +512,67 @@ def reconfigure(exit_on_success=true)
       end
     end
 
+    def check_license_acceptance(override_accept = false)
+      license_guard_file_path = File.join(data_path, ".license.accepted")
+
+      # If the project does not have a license we do not have
+      # any license to accept.
+      return unless File.exist?(project_license_path)
+
+      if !File.exist?(license_guard_file_path)
+        if override_accept || ask_license_acceptance
+          FileUtils.touch(license_guard_file_path)
+        else
+          puts "Please accept software's license in order to continue."
+          exit(1)
+        end
+      end
+    end
+
+    def ask_license_acceptance
+      puts "To use this software, you must agree to the terms of the software license agreement."
+
+      if !STDIN.tty?
+        puts "You need to either view and agree to the license agreement or pass --accept-license."
+        exit(1)
+      end
+
+      loop do
+        puts "Press 'Return' to view the software license agreement, 'q' to quit."
+
+        user_input = STDIN.getch
+        user_input << STDIN.getch while STDIN.ready?
+        case user_input
+        when "\n", "\r", "\r\n"
+          break
+        when "q"
+          return false
+        else
+          puts "Sorry, I do not understand '#{user_input}'."
+        end
+      end
+
+      system("less #{project_license_path}")
+
+      loop do
+        puts "Do you agree to the license agreement? [yes / no / q]"
+
+        user_input = gets.chomp
+        case user_input
+        when "yes"
+          return true
+        when "no", "q"
+          return false
+        else
+          puts "Sorry I do not understand '#{user_input}'."
```

Comment:
Actually - even simpler  would probably be "Type 'yes' to agree to the license agreement, or anything else to cancel." - because it's not important to us that they type exactly "no" so much as it is that they didn't type  "yes".